test/integration/apiserver/apply/apply_test.go

```diff
@@ -555,3 +557,257 @@ func TestApplyRemoveContainerPort(t *testing.T) {
+						"image": "nginx:latest"
+					}]
+				}
+			}
+		}
+	}`)
+	_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Body([]byte(obj)).Do().Get()
+	if err == nil {
+		t.Fatalf("Expecting to get version mismatch when applying object")
+	}
+	status, ok := err.(*errors.StatusError)
+	if !ok {
+		t.Fatalf("Expecting to get version mismatch as API error")
+	}
+	if status.Status().Code != http.StatusBadRequest {
+		t.Fatalf("expected status code to be %d but was %d", http.StatusBadRequest, status.Status().Code)
+	}
+}
+
+// TestApplyConvertsManagedFieldsVersion checks that the apply
+// converts the API group-version in the field manager
+func TestApplyConvertsManagedFieldsVersion(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	obj := []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+			"labels": {"app": "nginx"},
+			"managedFields": [
+				{
+					"manager": "create_test",
+					"operation": "Apply",
+					"apiVersion": "extensions/v1beta1",
+					"fields": {
+						"f:metadata": {
+							"f:labels": {
+								"f:app": {},
+								"f:track": {}
+							}
+						},
+						"f:spec": {
+							"f:labels": {
+								"f:app": {}
+							}
+						}
+					}
+				}
+			]
+		},
+		"spec": {
+			"replicas": 3,
+			"selector": {
+				"matchLabels": {
+					 "app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest"
+					}]
+				}
+			}
+		}
+	}`)
+
+	_, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Body(obj).Do().Get()
+	if err != nil {
+		t.Fatalf("Failed to create object using Apply patch: %v", err)
+	}
+
+	object, err := client.AppsV1().Deployments("default").Get("deployment", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	accessor, err := meta.Accessor(object)
+	if err != nil {
+		t.Fatalf("Failed to get meta accessor: %v", err)
+	}
+
+	managed := accessor.GetManagedFields()
+	if len(managed) != 1 && managed[0].Manager != "apply_test" {
+		t.Fatalf("Expected a single manager %v, but got managed fields: %v", "apply_test", managed)
+	}
+
+	if apiVersion := managed[0].APIVersion; apiVersion != "apps/v1" {
+		t.Fatalf("Expected to take ownership of resource with converted api version %v, but got %v", "apps/v1", apiVersion)
+	}
+
+	expected := []metav1.ManagedFieldsEntry{
+		{
+			Manager:    "apply_test",
```

Comment:
You have to create another entry for "create_test" to see that it still owns the "f:track" label